content/blog/gargle-1-2-0/index.Rmd

```diff
@@ -0,0 +1,243 @@
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+We're psyched to announce the release of [gargle](https://gargle.r-lib.org) 1.2.0.
+gargle is meant to take some of the pain out of working with Google APIs and is mostly aimed at the *maintainers* of R packages that call Google APIs.
+
+Packages that use gargle internally include [bigrquery](https://bigrquery.r-dbi.org), [gmailr](https://gmailr.r-lib.org), [googleAuthR](https://code.markedmondson.me/googleAuthR/), [googledrive](https://googledrive.tidyverse.org), and [googlesheets4](https://googlesheets4.tidyverse.org).
+Users of those packages, especially those who take an interest in the details around auth, may want to read on.
+
+You can install gargle from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("gargle")
+```
+
+I last blogged about gargle in [August 2019, around the release of v0.3.1](https://www.tidyverse.org/blog/2019/08/gargle-hello-world/).
+There have been several releases since then and this post highlights important changes across all of them.
+
+You can see a full list of changes in the [release notes](https://gargle.r-lib.org/news/index.html).
+
+```{r setup, include = FALSE}
+# TODO: probably delete this chunk; no one uses gargle directly
+library(gargle)
+```
+
+## Stewarding the cache of user tokens
+
+Users who "just go with the flow" generally find themselves doing the [OAuth Dance](https://medium.com/typeforms-engineering-blog/the-beginners-guide-to-oauth-dancing-4b8f3666de10) in the browser:
+
+```{r echo = FALSE, out.width = '100%', fig.cap = "Typical OAuth dance in the browser, when initiated from within R"}
+knitr::include_graphics("oauth-dance-web-flow.png")
+```
+
+Upon success, you see this message in the browser:
+
+```
+Authentication complete. Please close this page and return to R.
+```
+
+The dance concludes with gargle (or a wrapper package) in possession of a [Google user token](https://developers.google.com/identity/protocols/oauth2#installed).
+This credential usually comes with an access token, which is valid for about an hour, and a more persistent refresh token, which can be used to get new access tokens in the future.
+It is nice to store this refresh token somewhere, to reduce future friction for this user.
+We'll refer to this "somewhere" as the *oauth token cache*.
+
+One of the original motivations for the gargle package was to depart from httr's behaviour around token caching, which defaults to the current working directory.[^1]
+By default, gargle caches user tokens centrally, at the user level, and their keys or labels also convey which Google identity is associated with each token.
+This reduces the chance of accidentally sending tokens to remote servers (like DropBox or GitHub), increases the ability to reuse tokens across multiple R projects, and makes it easier to use multiple identities within a single project.
+
+[^1]: The successor to httr is under development and it will also cache user tokens at the user level, like gargle.
```

Comment:
I kinda want to allude to httr2, but without drawing undue, premature attention to it.

Reply:
I think it's fine to link to it; it's not a secret.